### PR TITLE
Autorun `make generate-syncsets` via GitHub actions

### PR DESCRIPTION
**Pull request by RiRa12621**
This adds GitHub actions that automatically run after you committed, to save users from that extra step

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ Folow a simple workflow:
 * Create Issue to explain what is wrong or missing
 * Fork this repository
 * Create a Pull Request referencing the issue you're adressing
+* enabled GitHub actions on your fork to automatically run generation steps
```
Comment (NautiluX) [inline CONTRIBUTING.md:7]: is this something people need to do manually?

Comment (RiRa12621) [reply CONTRIBUTING.md:7]: Iirc it depends on the defaults you have set

**Review by NautiluX — COMMENTED** [review]
/lgtm

**Pull request merged**